<a href="https://colab.research.google.com/github/SwapnilH09/Dataset-creation-Gangster_movies/blob/main/Gansta_movies_dataset_creation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## import libraries

In [1]:
## import necessary libraries
import requests
from bs4 import BeautifulSoup
import re
from tqdm import tqdm
import time

## load a single webpage and ETL its contents in a python dictionary

In [ ]:
## load the webpage
webpage = requests.get("https://en.wikipedia.org/wiki/Gangs_of_Wasseypur")

## get soup
soup = BeautifulSoup(webpage.content, 'html.parser')
content = soup.prettify()

#print(content)

In [ ]:
info_box = soup.find(class_="infobox vevent")
#print(info_box.prettify())

In [ ]:
info_rows = info_box.find_all("tr")
info_rows

In [10]:
def get_content_value(row_data):
    if row_data.find('li'):
        return [li.get_text(" ", strip=True).replace("\xa0", "") for li in row_data.find_all("li")]
    else:
        return row_data.get_text(" ", strip=True).replace("\xa0", "")

In [ ]:
movie_info = {}

for index, row in enumerate(info_rows):
    if index == 0:
        movie_info["title"] = row.find("th").get_text(" ", strip=True)
        #print(movie_info["title"])
    elif index == 1:
        continue
    else:
        content_key = row.find("th").get_text(" ", strip=True)
        content_val = get_content_value(row.find("td"))
        movie_info[content_key] = content_val
        #print(movie_info[content_key])

movie_info        

{'Box office': '₹ 1.540 billion',
 'Budget': '₹ 356 million',
 'Cinematography': 'Rajeev Ravi',
 'Country': 'India',
 'Directed by': 'Anurag Kashyap',
 'Distributed by': 'Viacom 18 Motion Pictures',
 'Edited by': 'Shweta Venkat',
 'Language': 'Hindi',
 'Music by': ['G. V. Prakash Kumar (score)', 'Sneha Khanwalkar (soundtrack)'],
 'Produced by': ['Atul Shukla', 'Anurag Kashyap', 'Sunil Bohra'],
 'Production companies': ['Anurag Kashyap Films', 'Jar Pictures'],
 'Release date': ['May2012 ( 2012-05 ) ( Cannes )',
  '22June2012 ( 2012-06-22 )'],
 'Running time': '321 minutes [1]',
 'Starring': ['Manoj Bajpayee',
  'Nawazuddin Siddiqui',
  'Richa Chadda',
  'Huma Qureshi',
  'Reema Sen',
  'Piyush Mishra',
  'Vineet Kumar Singh',
  'Pankaj Tripathi',
  'Tigmanshu Dhulia',
  'Sanjay Singh'],
 'Written by': ['Zeishan Quadri', 'Anurag Kashyap'],
 'title': 'Gangs of Wasseypur'}

## get info-box for all Gangster movies from https://en.wikipedia.org/wiki/Gangster_film

<b> Our goal in this part of the project is simple, extract the information in the info-box from wikipedia page of the movies. </b>

In [2]:
# load web-page
source_page = requests.get("https://en.wikipedia.org/wiki/Gangster_film")

# prepare soup
source_soup = BeautifulSoup(source_page.content, "html.parser")

In [ ]:
movies = source_soup.select("li a")
movies

In [ ]:
print(type(movies))
for idx, m in enumerate(movies[:5]):
    print(m.get("title"), m.get("href"))
    print(m)

We want to extract "title" and "link to their wiki" from this wikipedia page  https://en.wikipedia.org/wiki/Gangster_film
<br>
<b>Here, I will use regex to extract only those "href" which start with /wiki/ to avoid other links. </b>
<br>
This will filter some unwanted objects from our movies list[].

Task-list
<li> extract the links of movies' wikipedia page & their title </li>
<li> visit each link and collect information from the info-box on wikipedia page </li>
<li> store extracted info-box in a json file </li>


In [5]:
pattern = re.compile('\/wiki\/.*')  # pattern to match all href attributes starting with '/wiki/'

movies_doc = {}

base_path = "https://en.wikipedia.org"
for mov in movies:
    if pattern.match(str(mov.get("href"))):
        movies_doc[mov.get("title")] = base_path + mov.get("href")

In [6]:
movies_doc

{'3D film': 'https://en.wikipedia.org/wiki/3D_film',
 'A Better Tomorrow': 'https://en.wikipedia.org/wiki/A_Better_Tomorrow',
 "A Children's Story": 'https://en.wikipedia.org/wiki/A_Children%27s_Story',
 'A Very British Gangster': 'https://en.wikipedia.org/wiki/A_Very_British_Gangster',
 'A list of all special pages [q]': 'https://en.wikipedia.org/wiki/Special:SpecialPages',
 'A list of edits made from this IP address [y]': 'https://en.wikipedia.org/wiki/Special:MyContributions',
 'A list of recent changes to Wikipedia [r]': 'https://en.wikipedia.org/wiki/Special:RecentChanges',
 'Abhimanyu (1991 film)': 'https://en.wikipedia.org/wiki/Abhimanyu_(1991_film)',
 'Absolute film': 'https://en.wikipedia.org/wiki/Absolute_film',
 'Abstract animation': 'https://en.wikipedia.org/wiki/Abstract_animation',
 'Acid Western': 'https://en.wikipedia.org/wiki/Acid_Western',
 'Action film': 'https://en.wikipedia.org/wiki/Action_film',
 'Actuality film': 'https://en.wikipedia.org/wiki/Actuality_film',
 '

The dictionary movie_doc contains our movies' titles and links to their wikipedia websites. But it also contains some of the garbage, for eg. the first 5 elements of the dictionary are no links to any movies. 
<br>
In this project, we only need to extract the info-box of gangster movies, so we will visit every page via their specific links, and if they do not contain information of our choice, we will discard that entry from our movies_doc dictionary.

In [7]:
len(movies_doc)

611

In [13]:
def get_content_value(row_data):
    if row_data.find('li'):
        return [li.get_text(" ", strip=True).replace("\xa0", "") for li in row_data.find_all("li")]
    else:
        return row_data.get_text(" ", strip=True).replace("\xa0", "")


def get_info_box(url):
    """ this function will extract the infobox-vevent from the url """

    r = requests.get(url)
    soup = BeautifulSoup(r.content, "html.parser")

    info_box = soup.find(class_="infobox vevent")
    info_rows = info_box.find_all("tr")

    movie_info = {}

    for idx, row in enumerate(info_rows):
        if idx == 0:
            movie_info["title"] = row.find("th").get_text(" ", strip=True)
        elif idx == 1:
            continue
        else:
            content_key = row.find("th").get_text(" ", strip=True)
            content_val = get_content_value(row.find("td"))
            movie_info[content_key] = content_val
    
    return movie_info

In [16]:
movie_info_list = []
for idx, v in tqdm(enumerate(movies_doc.values())):

    # put program to sleep, in order to not exploit wikipedia against its rules, as defined in its robot.txt file
    # https://en.wikipedia.org/robots.txt <- link for robot.txt file for wiki
    if idx%10==0:
        time.sleep(10.0)

    try:
        movie_info_list.append(get_info_box(v))
    except Exception as e:
        print(e, v)    


0it [00:00, ?it/s]
1it [00:10, 10.17s/it]
2it [00:10,  7.16s/it]
4it [00:10,  5.04s/it]
5it [00:10,  3.57s/it]
7it [00:10,  2.52s/it]
9it [00:10,  1.78s/it]
11it [00:21,  2.77s/it]
13it [00:21,  1.96s/it]
16it [00:21,  1.38s/it]
19it [00:21,  1.02it/s]
21it [00:31,  2.21s/it]
24it [00:31,  1.56s/it]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Caliber_9



25it [00:31,  1.14s/it]
28it [00:31,  1.23it/s]

'NoneType' object has no attribute 'find' https://en.wikipedia.org/wiki/Il_Capo_dei_Capi



30it [00:32,  1.71it/s]
32it [00:42,  1.93s/it]
33it [00:42,  1.38s/it]
36it [00:42,  1.02it/s]
38it [00:42,  1.42it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Fort_Apache_Napoli



40it [00:42,  1.95it/s]
42it [00:52,  1.87s/it]
43it [00:52,  1.35s/it]
45it [00:53,  1.04it/s]
48it [00:53,  1.45it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/The_Italian_Connection
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/The_Legendary_Giulia_and_Other_Miracles
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/The_Mafia_Kills_Only_in_Summer_(film)



51it [01:03,  1.50s/it]
53it [01:03,  1.06s/it]
55it [01:03,  1.31it/s]
58it [01:03,  1.83it/s]
60it [01:03,  2.45it/s]

'NoneType' object has no attribute 'find' https://en.wikipedia.org/wiki/La_piovra



62it [01:13,  1.80s/it]
63it [01:14,  1.30s/it]
66it [01:14,  1.08it/s]
69it [01:14,  1.51it/s]
71it [01:24,  1.98s/it]
73it [01:24,  1.40s/it]
75it [01:24,  1.00it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/The_Immortal_(2019_film)



78it [01:24,  1.40it/s]
80it [01:25,  1.89it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Kray_Twins#In_popular_culture



82it [01:35,  1.89s/it]
83it [01:35,  1.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/The_Rise_of_the_Krays



85it [01:35,  1.00it/s]
87it [01:35,  1.39it/s]
89it [01:35,  1.92it/s]
91it [01:46,  1.89s/it]
93it [01:46,  1.34s/it]
95it [01:46,  1.03it/s]
97it [01:46,  1.43it/s]
99it [01:46,  1.97it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/A_Very_British_Gangster



101it [01:56,  1.89s/it]
103it [01:56,  1.34s/it]
105it [01:57,  1.05it/s]
107it [01:57,  1.46it/s]
109it [01:57,  1.83it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Dacoit_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Dacoit



111it [02:07,  1.94s/it]
112it [02:08,  1.46s/it]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Gunga_Jumna



113it [02:08,  1.09s/it]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Bandit_Queen



115it [02:09,  1.18it/s]
116it [02:09,  1.56it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mumbai_underworld_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mumbai_underworld



117it [02:09,  1.83it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Slum



118it [02:10,  1.65it/s]
120it [02:10,  2.22it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mumbai
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Haji_Mastan
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Dawood_Ibrahim



121it [02:20,  3.34s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/D-Company



123it [02:20,  2.37s/it]
124it [02:21,  1.80s/it]
125it [02:21,  1.30s/it]

'NoneType' object has no attribute 'find' https://en.wikipedia.org/wiki/Don_(franchise)



126it [02:21,  1.03it/s]
128it [02:21,  1.41it/s]
129it [02:22,  1.74it/s]
130it [02:22,  1.83it/s]
131it [02:32,  3.44s/it]
132it [02:32,  2.44s/it]
133it [02:32,  1.74s/it]
134it [02:33,  1.33s/it]
135it [02:33,  1.03it/s]
136it [02:33,  1.35it/s]
137it [02:33,  1.75it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mafia_Raj



139it [02:34,  2.13it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Neo_noir



141it [02:44,  1.85s/it]
142it [02:44,  1.37s/it]
144it [02:44,  1.02it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Selvaraghavan
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Rajeev_Ravi



145it [02:45,  1.36it/s]
146it [02:45,  1.80it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Vikram_Vedha



147it [02:45,  2.04it/s]
149it [02:45,  2.74it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Vetrimaran



150it [02:46,  2.78it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Kannada



151it [02:56,  3.29s/it]
152it [02:56,  2.35s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gun_fu
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Heroic_bloodshed



154it [02:56,  1.66s/it]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/A_Better_Tomorrow



155it [02:56,  1.26s/it]
156it [02:56,  1.08it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/The_Killer_(1989_film)
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/To_Be_Number_One_(film)



157it [02:57,  1.44it/s]
159it [02:57,  1.98it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Infernal_Affairs



161it [03:07,  1.87s/it]
163it [03:07,  1.35s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gangster_film



164it [03:07,  1.02s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Portal:Film



165it [03:08,  1.27it/s]
166it [03:08,  1.71it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Doi_(identifier)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/JSTOR_(identifier)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/I.B._Tauris



168it [03:08,  2.20it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/ISBN_(identifier)



169it [03:09,  1.57it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/978-1-84885-040-8



170it [03:10,  1.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/9780190264079



171it [03:21,  3.78s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/978-0-7864-3136-6



172it [03:22,  2.90s/it]
173it [03:22,  2.08s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/978-0-415-28132-4
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Rotten_Tomatoes



174it [03:22,  1.55s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Fandango_(company)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/IndieWire



176it [03:22,  1.13s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/OCLC_(identifier)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:CS1_maint:_multiple_names:_authors_list



178it [03:23,  1.07it/s]
179it [03:24,  1.42it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/978-0-8108-7541-8
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mark_Kermode
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Guardian_Unlimited



181it [03:35,  2.13s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:BookSources/978-1-4051-6370-5



182it [03:35,  1.61s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Template:Film_genres
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Template_talk:Film_genres



184it [03:35,  1.16s/it]
185it [03:35,  1.17it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Action_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Arthouse_action_film



186it [03:36,  1.51it/s]
187it [03:36,  2.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hong_Kong_action_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Adventure_film



188it [03:36,  2.35it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Survival_film



189it [03:36,  2.76it/s]
190it [03:36,  3.40it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Art_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Biographical_film



191it [03:46,  3.25s/it]
192it [03:47,  2.33s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Christian_film_industry
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Comedy_film



193it [03:47,  1.69s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Black_comedy



194it [03:47,  1.26s/it]
195it [03:47,  1.08it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Commedia_all%27italiana
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Commedia_sexy_all%27italiana



196it [03:47,  1.46it/s]
197it [03:47,  1.94it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Bromantic_comedy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Comedy-drama



198it [03:48,  2.52it/s]
199it [03:48,  3.17it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gross_out
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Comedy_horror



200it [03:48,  3.21it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Parody_film



201it [03:58,  3.26s/it]
202it [03:58,  2.32s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mo_lei_tau
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Thriller_(genre)



203it [03:58,  1.66s/it]
204it [03:58,  1.20s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Comedy_of_remarriage
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Romantic_comedy



205it [03:59,  1.03it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sex_comedy



206it [03:59,  1.27it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Screwball_comedy_film



207it [03:59,  1.62it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Silent_comedy



208it [04:00,  2.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Slapstick_film



209it [04:00,  2.26it/s]
210it [04:00,  2.85it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cyberpunk
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Japanese_cyberpunk



211it [04:10,  3.34s/it]
212it [04:11,  2.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Documentary_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Animated_documentary



213it [04:11,  1.70s/it]
214it [04:11,  1.24s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Docudrama
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mockumentary



215it [04:11,  1.09it/s]
216it [04:11,  1.48it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mondo_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pseudo-documentary



217it [04:11,  1.87it/s]
218it [04:11,  2.44it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Semidocumentary
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Travel_documentary



219it [04:12,  2.99it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Drama_(film_and_television)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Calligrafismo



221it [04:22,  1.76s/it]
222it [04:22,  1.28s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Historical_drama
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Legal_drama



223it [04:22,  1.06it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Melodrama#Film



224it [04:23,  1.28it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Korean_melodrama



225it [04:23,  1.48it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sex_in_film



226it [04:23,  1.85it/s]
227it [04:23,  2.40it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pink_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sexploitation_film



228it [04:23,  2.95it/s]
229it [04:24,  3.62it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Erotic_thriller
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Educational_film



230it [04:24,  4.44it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Social_guidance_film



231it [04:34,  3.23s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Epic_film



232it [04:34,  2.36s/it]
233it [04:34,  1.70s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sword-and-sandal
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Experimental_film



234it [04:35,  1.26s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Exploitation_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Template:Exploitation_film



236it [04:35,  1.07it/s]
237it [04:35,  1.43it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Fantasy_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Fantasy_comedy



238it [04:35,  1.84it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Contemporary_fantasy



239it [04:36,  2.11it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Fairy_tale#Film



240it [04:36,  2.35it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Fantastique



241it [04:46,  3.34s/it]
242it [04:46,  2.39s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/High_fantasy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Historical_fantasy



243it [04:47,  1.76s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Magic_realism



244it [04:47,  1.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Science_fantasy



245it [04:47,  1.00s/it]
246it [04:47,  1.35it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Film_noir
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Neo-noir



248it [04:47,  1.81it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pulp_noir
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Tech_noir



249it [04:48,  2.16it/s]
250it [04:48,  2.72it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gothic_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gothic_romance_film



251it [04:58,  3.31s/it]
252it [04:58,  2.36s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Southern_Gothic#Film_and_television
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gothic_science_fiction



253it [04:58,  1.70s/it]
254it [04:59,  1.23s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Suburban_Gothic
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Urban_Gothic#Film



255it [04:59,  1.02s/it]
256it [04:59,  1.34it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Horror_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Art_horror



257it [04:59,  1.73it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_body_horror_media



258it [05:00,  1.98it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cannibal_film



259it [05:00,  2.40it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Chinese_horror_film



260it [05:00,  2.72it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_eco-horror_films



261it [05:10,  3.32s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Found_footage_(film_technique)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/German_underground_horror



263it [05:11,  2.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_ghost_films



264it [05:11,  1.78s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Giallo



265it [05:11,  1.33s/it]
266it [05:12,  1.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Holiday_horror
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Japanese_horror



267it [05:12,  1.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Korean_horror



268it [05:12,  1.60it/s]
269it [05:12,  2.04it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Lovecraftian_horror



270it [05:12,  2.42it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_natural_horror_films



271it [05:23,  3.34s/it]
272it [05:23,  2.40s/it]
273it [05:23,  1.73s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Psycho-biddy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Psychological_horror
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Religious_horror_films



275it [05:23,  1.25s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_science_fiction_horror_films



276it [05:24,  1.03it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Slasher_film



277it [05:24,  1.33it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Splatter_film



278it [05:24,  1.64it/s]
279it [05:24,  2.14it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Satanic_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Maximalist_film



280it [05:24,  2.72it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Minimalist_film



281it [05:35,  3.35s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Musical_film



282it [05:35,  2.46s/it]
283it [05:35,  1.78s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Backstage_musical
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Jukebox_musical



284it [05:36,  1.34s/it]
285it [05:36,  1.03it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Musicarello
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Operetta_film



286it [05:36,  1.30it/s]


'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sceneggiata
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mystery_film


287it [05:36,  1.67it/s]
288it [05:37,  1.96it/s]
289it [05:37,  2.41it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Detective_fiction
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Occult_detective_fiction



290it [05:37,  3.10it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Whodunit



291it [05:47,  3.27s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pop_culture_fiction



292it [05:47,  2.35s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pornographic_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hardcore_pornography



294it [05:47,  1.69s/it]
295it [05:48,  1.22s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Softcore_pornography
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Propaganda_film



296it [05:48,  1.12it/s]
297it [05:48,  1.49it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Reality_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Romance_film



298it [05:48,  1.74it/s]
299it [05:48,  2.29it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Romantic_fantasy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Paranormal_romance



300it [05:49,  2.56it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Romantic_thriller



301it [05:59,  3.38s/it]
302it [05:59,  2.40s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Science_fiction_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Arthouse_science_fiction_film



303it [05:59,  1.79s/it]
304it [06:00,  1.30s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Science_fiction_comedy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Military_science_fiction



305it [06:00,  1.02it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/New_Wave_science_fiction



306it [06:00,  1.28it/s]
307it [06:00,  1.66it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Parallel_universe_(fiction)#Films
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Planetary_romance



308it [06:00,  2.14it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Space_opera



309it [06:01,  2.16it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Steampunk



310it [06:01,  2.49it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Tokusatsu



311it [06:11,  3.32s/it]
312it [06:11,  2.36s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Space_Western
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Slice_of_life#Cinema



313it [06:12,  1.71s/it]
314it [06:12,  1.25s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Slow_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Thriller_film



315it [06:12,  1.10it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Comedy_thriller



316it [06:12,  1.30it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Conspiracy_fiction#Film_and_television
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Financial_thriller



318it [06:13,  1.73it/s]
319it [06:13,  2.27it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Legal_thriller
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Political_thriller



320it [06:13,  2.76it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Psychological_thriller



321it [06:23,  3.30s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Techno-thriller



322it [06:23,  2.40s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Transgressive_art



323it [06:24,  1.82s/it]
325it [06:24,  1.30s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cinema_of_Transgression
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Extreme_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Trick_film



326it [06:24,  1.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_about_animals



327it [06:25,  1.32it/s]
328it [06:25,  1.77it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Beach_party_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Body_swap



329it [06:25,  1.82it/s]
330it [06:25,  2.41it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Buddy_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Buddy_cop_film



331it [06:35,  3.32s/it]
332it [06:35,  2.37s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Female_buddy_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Chicano_films



333it [06:36,  1.69s/it]
334it [06:36,  1.23s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Colonial_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Coming-of-age_story



335it [06:36,  1.11it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Concert_film



336it [06:36,  1.43it/s]
337it [06:36,  1.85it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Crime_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gentleman_thief



338it [06:37,  2.05it/s]
339it [06:37,  2.62it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gokud%C5%8D
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gong%27an_fiction



340it [06:37,  3.16it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Heist_film



341it [06:47,  3.25s/it]
342it [06:47,  2.32s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hood_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mafia_film



343it [06:47,  1.69s/it]
344it [06:48,  1.22s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mafia_comedy



345it [06:48,  1.11it/s]
346it [06:48,  1.48it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Yakuza_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Dance_in_film



347it [06:48,  1.88it/s]
348it [06:48,  2.35it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Disaster_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_apocalyptic_films



349it [06:49,  2.53it/s]
350it [06:49,  3.16it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_drug_films
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Psychedelic_film



351it [06:59,  3.28s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Stoner_film



352it [06:59,  2.44s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_dystopian_films



353it [07:00,  1.82s/it]
354it [07:00,  1.33s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Ecchi
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Economics_film



355it [07:00,  1.03it/s]
356it [07:00,  1.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Ethnographic_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Blaxploitation



357it [07:01,  1.67it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mexploitation



358it [07:01,  2.04it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Turksploitation



359it [07:01,  1.93it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_featuring_extraterrestrials



360it [07:02,  2.26it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_about_food_and_drink



361it [07:12,  3.40s/it]
362it [07:12,  2.42s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Funny_animal
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gendai-geki



363it [07:12,  1.73s/it]
364it [07:12,  1.25s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Goona-goona_epic
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Girls_with_guns



365it [07:12,  1.09it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Harem_(genre)



366it [07:13,  1.38it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hentai



367it [07:13,  1.64it/s]
368it [07:13,  2.09it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Lolicon
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Shotacon



369it [07:13,  2.63it/s]
370it [07:13,  3.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Tentacle_erotica
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Heimatfilm



371it [07:24,  3.27s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Isekai



372it [07:24,  2.38s/it]
373it [07:24,  1.72s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Jidaigeki
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Samurai_cinema



374it [07:25,  1.41s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_LGBT-related_films



375it [07:25,  1.13s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Yaoi



376it [07:26,  1.09it/s]
377it [07:26,  1.49it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Yuri_(genre)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Luchador_films



378it [07:26,  1.79it/s]
379it [07:26,  2.30it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Magical_girl
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Martial_arts_film



380it [07:27,  2.57it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Bruceploitation



381it [07:37,  3.34s/it]
382it [07:37,  2.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Chopsocky
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Kung_fu_film



383it [07:37,  1.77s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_ninja_films



384it [07:38,  1.32s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wuxia#Films



385it [07:38,  1.01s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mecha



386it [07:38,  1.31it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mecha_anime_and_manga
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Monster_movie



388it [07:38,  1.69it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_featuring_giant_monsters



389it [07:39,  2.09it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Jiangshi_fiction#Hong_Kong_cinema



390it [07:39,  2.24it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Kaiju



391it [07:49,  3.36s/it]
392it [07:49,  2.40s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_mummy_films
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Vampire_films



393it [07:50,  1.76s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Werewolf_fiction#Films



394it [07:50,  1.43s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_zombie_films



395it [07:50,  1.08s/it]
396it [07:51,  1.26it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Zombie_comedy
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mountain_film



397it [07:51,  1.49it/s]
398it [07:51,  1.98it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cinema_da_Boca_do_Lixo
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Muslim_social



399it [07:52,  2.10it/s]
400it [07:52,  2.56it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Nature_documentary
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_environmental_films



401it [08:02,  3.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Opera_film



402it [08:02,  2.39s/it]
403it [08:02,  1.72s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Outlaw_biker_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Ozploitation



404it [08:02,  1.24s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Partisan_film
'NoneType' object has no attribute 'find_all'


405it [08:03,  1.07it/s]
406it [08:03,  1.45it/s]

 https://en.wikipedia.org/wiki/Pirates_in_popular_culture#Films
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Prison_film



407it [08:03,  1.90it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Women_in_prison_film



408it [08:03,  2.31it/s]
409it [08:03,  2.93it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Race_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Rape_and_revenge_film



410it [08:03,  3.17it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Road_movie



411it [08:14,  3.25s/it]
412it [08:14,  2.32s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Tr%C3%BCmmerfilm
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Rumberas_film



413it [08:14,  1.72s/it]
414it [08:14,  1.24s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Bavarian_porn
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mexican_sex_comedy



415it [08:14,  1.06it/s]
416it [08:14,  1.44it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Nazi_exploitation
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pornochanchada



417it [08:15,  1.90it/s]
418it [08:15,  2.50it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Nunsploitation
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sex_report_film



419it [08:15,  3.18it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Shomin-geki



420it [08:15,  3.51it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_featuring_slavery



421it [08:25,  3.24s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Snuff_film



422it [08:25,  2.36s/it]
423it [08:26,  1.69s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/South_Seas_genre
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sports_film



424it [08:26,  1.23s/it]
425it [08:26,  1.11it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Spy_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Eurospy_film



426it [08:26,  1.44it/s]
427it [08:26,  1.90it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Superhero_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Surf_film



428it [08:26,  2.28it/s]
429it [08:27,  2.79it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Swashbuckler_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sword_and_sorcery



430it [08:27,  3.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Trial_film



431it [08:37,  3.27s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Vigilante_film



432it [08:37,  2.42s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/War_film



433it [08:38,  1.78s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_anti-war_films



434it [08:38,  1.32s/it]
435it [08:38,  1.02it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Euro_War
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Submarine_films



436it [08:39,  1.22it/s]
437it [08:39,  1.65it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Western_(genre)
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Acid_Western



438it [08:39,  2.19it/s]
439it [08:39,  2.79it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Florida_Western
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Meat_pie_Western



440it [08:39,  3.10it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Northern_(genre)



441it [08:49,  3.30s/it]
442it [08:50,  2.36s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Ostern
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Revisionist_Western



443it [08:50,  1.74s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Spaghetti_Western



444it [08:50,  1.28s/it]
445it [08:50,  1.05it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Weird_West
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Absolute_film



446it [08:50,  1.38it/s]
447it [08:51,  1.79it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/New_Objectivity_(filmmaking)



448it [08:51,  2.12it/s]
449it [08:51,  2.61it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Australian_New_Wave
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Auteur



450it [08:51,  3.12it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Berlin_School_(filmmaking)



451it [09:01,  3.26s/it]
452it [09:02,  2.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Bourekas_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Brighton_School_(filmmaking)



453it [09:02,  1.67s/it]
454it [09:02,  1.21s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/British_New_Wave
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Kitchen_sink_realism



455it [09:02,  1.14it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Budapest_school



456it [09:02,  1.47it/s]
457it [09:02,  1.85it/s]
458it [09:03,  2.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cin%C3%A9ma_pur
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Czechoslovak_New_Wave



460it [09:03,  2.57it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Documentary_Film_Movement



461it [09:13,  3.31s/it]
462it [09:13,  2.42s/it]
463it [09:14,  1.73s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Erra_Cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/European_art_cinema



464it [09:14,  1.27s/it]
465it [09:14,  1.07it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Film_gris
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Free_Cinema



466it [09:14,  1.42it/s]
467it [09:14,  1.89it/s]
468it [09:14,  2.36it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/German_Expressionism
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Golden_Age_of_Nigerian_Cinema



469it [09:15,  3.02it/s]
470it [09:15,  3.73it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Grupo_Cine_Liberaci%C3%B3n
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hollywood_on_the_Tiber



471it [09:25,  3.27s/it]
472it [09:25,  2.32s/it]
473it [09:25,  1.66s/it]
474it [09:25,  1.20s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Italian_futurism_in_cinema
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Italian_neorealism



475it [09:25,  1.13it/s]
476it [09:26,  1.38it/s]
477it [09:26,  1.84it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Kammerspielfilm



478it [09:26,  2.14it/s]
479it [09:26,  2.59it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Lettrism
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Modernist_film



480it [09:27,  2.71it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Neorealism_(art)



481it [09:37,  3.29s/it]
482it [09:37,  2.41s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/New_Generation_(Malayalam_film_movement)



483it [09:38,  1.84s/it]
484it [09:38,  1.36s/it]
485it [09:38,  1.01it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/New_Nigerian_Cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/New_Queer_Cinema



486it [09:38,  1.37it/s]
487it [09:39,  1.64it/s]
488it [09:39,  2.04it/s]
489it [09:39,  2.51it/s]
490it [09:39,  2.66it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Persian_Film



491it [09:49,  3.29s/it]
492it [09:49,  2.34s/it]
493it [09:50,  1.67s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Polish_Film_School
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pra%C5%A1ka_filmska_%C5%A1kola



494it [09:50,  1.21s/it]
495it [09:50,  1.14it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Prussian_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Pure_Film_Movement



496it [09:50,  1.46it/s]
497it [09:50,  1.92it/s]

'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Remodernist_film



498it [09:50,  2.29it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Socialist_realism



499it [09:51,  2.65it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Social_realism



500it [09:51,  2.75it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Soviet_Parallel_Cinema



501it [10:01,  3.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Structural_film



502it [10:01,  2.39s/it]
504it [10:02,  1.70s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Surrealist_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Telefoni_Bianchi
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Third_Cinema



506it [10:02,  1.22s/it]
507it [10:02,  1.13it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Vulgar_auteurism



508it [10:02,  1.36it/s]
509it [10:02,  1.77it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Black_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Children%27s_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Children%27s_anime_and_manga



511it [10:13,  1.92s/it]
512it [10:13,  1.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Seinen_manga
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Stag_film



513it [10:13,  1.01s/it]
514it [10:13,  1.36it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Teen_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sh%C5%8Dnen_manga



515it [10:13,  1.68it/s]
516it [10:13,  2.17it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sh%C5%8Djo_manga
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Woman%27s_film



517it [10:14,  2.63it/s]
518it [10:14,  3.32it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Chick_flick
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Josei_manga



519it [10:14,  3.40it/s]
520it [10:14,  4.12it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/3D_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Actuality_film



521it [10:24,  3.28s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Animation



522it [10:25,  2.40s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Anime



523it [10:25,  1.79s/it]
524it [10:25,  1.30s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Animated_cartoon
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Arthouse_animation



525it [10:26,  1.02it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Computer_animation



526it [10:26,  1.33it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Stop_motion



527it [10:26,  1.70it/s]
528it [10:26,  2.25it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Traditional_animation
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Anthology_film



529it [10:27,  2.30it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/B_movie



530it [10:27,  2.24it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Black_and_white



531it [10:37,  3.35s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Blockbuster_(entertainment)



532it [10:38,  2.53s/it]
533it [10:38,  1.81s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Bollywood
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Masala_film



534it [10:38,  1.31s/it]
535it [10:38,  1.03it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cin%C3%A9ma_v%C3%A9rit%C3%A9
'NoneType' object has no attribute 'get_text' https://en.wikipedia.org/wiki/Classical_Hollywood_cinema



536it [10:39,  1.29it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Collage_film



537it [10:39,  1.61it/s]
538it [10:39,  2.13it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Color_motion_picture_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Compilation_film



539it [10:39,  2.74it/s]
540it [10:39,  3.46it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Composite_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Computer_screen_film



541it [10:49,  3.30s/it]
542it [10:50,  2.36s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Cult_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Midnight_movie



543it [10:50,  1.71s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Database_cinema



544it [10:50,  1.31s/it]
545it [10:50,  1.04it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Docufiction
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Ethnofiction



546it [10:51,  1.25it/s]
547it [10:51,  1.65it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Abstract_animation
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Feature_film



548it [10:51,  1.88it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Featurette



549it [10:52,  2.23it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Film_%C3%A0_clef



550it [10:52,  2.61it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Film-poem



551it [11:02,  3.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Grindhouse



552it [11:02,  2.44s/it]
553it [11:03,  1.77s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Hyperlink_cinema
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Independent_film



554it [11:03,  1.28s/it]
555it [11:03,  1.07it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Guerrilla_filmmaking
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Interstitial_art



556it [11:03,  1.46it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Live_action



557it [11:03,  1.71it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/List_of_films_with_live_action_and_animation



558it [11:04,  1.93it/s]
559it [11:04,  2.42it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Lollywood
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Low-budget_film



560it [11:04,  2.05it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Major_film_studio



561it [11:15,  3.38s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Making-of



562it [11:15,  2.47s/it]
563it [11:15,  1.76s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Message_picture
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Metacinema



564it [11:15,  1.29s/it]
565it [11:15,  1.07it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mockbuster
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Musical_short



566it [11:16,  1.35it/s]
567it [11:16,  1.78it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Mythopoeia
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/No-budget_film



568it [11:16,  2.22it/s]
569it [11:16,  2.88it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/One-shot_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Paracinema



570it [11:16,  2.95it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Participatory_cinema



571it [11:27,  3.34s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Poetry_film



572it [11:27,  2.41s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Postmodernist_film



573it [11:27,  1.75s/it]
574it [11:27,  1.27s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Reverse_motion
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Satire_(film_and_television)



575it [11:27,  1.06it/s]
576it [11:28,  1.44it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Serial_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Shinpa



577it [11:28,  1.92it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Short_film



578it [11:28,  2.31it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Silent_film



579it [11:28,  2.39it/s]
580it [11:28,  3.05it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Sound_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Underground_film



581it [11:39,  3.28s/it]
582it [11:39,  2.33s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Video_nasty
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Z_movie
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Gangster_films



584it [11:39,  1.66s/it]
586it [11:39,  1.18s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Film_genres
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Articles_with_short_description
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Short_description_matches_Wikidata



587it [11:39,  1.04it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Articles_containing_Russian-language_text



588it [11:40,  1.33it/s]
590it [11:40,  1.81it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Commons_category_link_is_on_Wikidata
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Wikipedia_articles_with_GND_identifiers
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Wikipedia_articles_with_BNF_identifiers



591it [11:50,  3.42s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Category:Wikipedia_articles_with_LCCN_identifiers



592it [11:50,  2.47s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:MyTalk



593it [11:51,  1.84s/it]
594it [11:51,  1.34s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:MyContributions
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Gangster_film



595it [11:51,  1.03s/it]
597it [11:51,  1.35it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Talk:Gangster_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Main_Page
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:Contents



598it [11:52,  1.61it/s]
599it [11:52,  2.11it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Portal:Current_events
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:Random



600it [11:52,  2.51it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:About



601it [12:02,  3.31s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Help:Contents
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Help:Introduction



603it [12:02,  2.38s/it]
604it [12:03,  1.71s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:Community_portal
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:RecentChanges



605it [12:03,  1.27s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:File_Upload_Wizard
'NoneType' object has no attribute 'find_all' 


606it [12:03,  1.06it/s]

https://en.wikipedia.org/wiki/Special:WhatLinksHere/Gangster_film



607it [12:04,  1.13it/s]
608it [12:04,  1.53it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:RecentChangesLinked/Gangster_film
'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:File_Upload_Wizard



609it [12:04,  1.86it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Special:SpecialPages



610it [12:04,  2.28it/s]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:About



611it [12:14,  1.20s/it]

'NoneType' object has no attribute 'find_all' https://en.wikipedia.org/wiki/Wikipedia:General_disclaimer


In [17]:
len(movie_info_list)

148

In [18]:
movie_info_list[:10]

[{'Based on': 'The Return of Superfly by Mark Jacobson',
  'Box office': '$266.5 million [1]',
  'Budget': '$100 million [1]',
  'Cinematography': 'Harris Savides',
  'Country': 'United States',
  'Directed by': 'Ridley Scott',
  'Distributed by': 'Universal Pictures',
  'Edited by': 'Pietro Scalia',
  'Language': 'English',
  'Music by': 'Marc Streitenfeld',
  'Produced by': ['Ridley Scott', 'Brian Grazer'],
  'Production companies': ['Imagine Entertainment',
   'Scott Free Productions',
   'Relativity Media'],
  'Release date': ['October19,2007 ( 2007-10-19 ) ( Apollo Theater )',
   'November2,2007 ( 2007-11-02 ) (United States)'],
  'Running time': '157 minutes',
  'Screenplay by': 'Steven Zaillian',
  'Starring': ['Russell Crowe',
   'Denzel Washington',
   'Chiwetel Ejiofor',
   'Cuba Gooding Jr.',
   'Josh Brolin',
   'Ted Levine',
   'Armand Assante',
   'John Ortiz',
   'John Hawkes',
   'RZA'],
  'title': 'American Gangster'},
 {'Based on': 'Shaft by Ernest Tidyman',
  'Box of

<b> Important points to notice after observing the results </b>
<li> We collect total 611 links, but ended up only scraping data for 148 movies. This happened because most the links we either not for a movie or they did not have a infobox-vevent section </li>

## Save Movie data

In [19]:
import json

In [22]:
def save_movies_data(title, data):
    """ use this function to save python dictionary as json file """ 
    with open(title, "w", encoding="utf-8") as f:
        json.dump(data, f, ensure_ascii=False, indent=4)

In [21]:
def load_data(title):
    """ use this function to load json file as python dictionary """
    with open(title, encoding="utf-8") as f:
        return json.load(f)

In [23]:
save_movies_data("Ganster_movies_data.json", movie_info_list)